core/src/navigation/LeftNav.html

```diff
@@ -213,9 +289,82 @@
     },
     methods: {
       handleClick(node) {
+        if (this.getIsSemiCollapsed()) {
+          this.closePopupMenu();
+        }
         this.fire('handleClick', { node });
       },
 
+      handleIconClick(nodeOrNodes, el) {
+        if (this.getIsSemiCollapsed()) {
+          event.stopPropagation();
+
+          let selectedCategory;
+          let sideBar = document.getElementsByClassName('fd-app__sidebar')[0];
+
+          if (nodeOrNodes.metaInfo && nodeOrNodes.metaInfo.label) {
+            selectedCategory = nodeOrNodes.metaInfo.label;
+          } else {
+            selectedCategory =
+              (nodeOrNodes.category && nodeOrNodes.category.label) ||
+              nodeOrNodes.category;
+          }
+
+          if (!sideBar.classList.contains('isBlocked')) {
+            sideBar.className += ' isBlocked';
+          }
+
+          // only close if same clicked
+          if (selectedCategory === this.get().selectedCategory) {
+            this.closePopupMenu();
+            return;
+          }
+
+          this.set({
+            selectedCategory
+          });
+
+          this.calculateFlyoutPosition(el);
+        }
+      },
+
+      calculateFlyoutPosition(el) {
+        //Calculate top/bottom position for flyout sublist
+        let parent = el.closest('.fd-side-nav__group');
+        let parentTopPosition = parent.offsetTop;
+        let shellbarHeight = Luigi.elements().getShellbar().offsetHeight;
```

Comment:
Internally we should not rely on the `window.Luigi` object, since you already added `import { LuigiElements } from '../core-api';` you can use that one.

Reply:
good point! done)